validator-cli/src/utils/arbMsgExecutor.ts

```diff
@@ -0,0 +1,87 @@
+import { ContractTransaction } from "@ethersproject/contracts";
+
+// Execute the child-to-parent (arbitrum-to-ethereum) message, for reference see: https://docs.arbitrum.io/sdk/reference/message/ChildToParentMessage
+async function messageExecutor(trnxHash: string, childRpc: string, parentRpc: string): Promise<ContractTransaction> {
+  const PRIVATE_KEY = process.env.PRIVATE_KEY;
+  const childJsonRpc = new JsonRpcProvider(childRpc);
+  const childProvider = new ArbitrumProvider(childJsonRpc);
+  const parentProvider = new JsonRpcProvider(parentRpc);
+
+  let childReceipt: TransactionReceipt | null;
+  try {
+    childReceipt = await childProvider.getTransactionReceipt(trnxHash);
+  } catch (error) {
+    throw new Error(`Failed to get child transaction receipt: ${error.message}`);
+  }
+  if (!childReceipt) {
+    throw new Error(`Transaction receipt not found for hash: ${trnxHash}`);
+  }
+
+  const messageReceipt = new ChildTransactionReceipt(childReceipt);
+  const parentSigner: Signer = new Wallet(PRIVATE_KEY, parentProvider);
+
+  let childToParentMessage: ChildToParentMessageWriter;
+  try {
+    const messages = await messageReceipt.getChildToParentMessages(parentSigner);
+    childToParentMessage = messages[0];
+    if (!childToParentMessage) {
+      throw new Error("No child-to-parent messages found");
+    }
+  } catch (error) {
+    throw new Error(`Failed to retrieve child-to-parent messages: ${error.message}`);
+  }
+
+  // Execute the message
+  try {
+    const res = await childToParentMessage.execute(childProvider);
+    return res;
+  } catch (error) {
+    throw new Error(`Message execution failed: ${error.message}`);
+  }
+}
+
+async function getMessageStatus(
+  trnxHash: string,
+  childRpc: string,
+  parentRpc: string
+): Promise<ChildToParentMessageStatus> {
+  const PRIVATE_KEY = process.env.PRIVATE_KEY;
+  const childJsonRpc = new JsonRpcProvider(childRpc);
+  const childProvider = new ArbitrumProvider(childJsonRpc);
+  const parentProvider = new JsonRpcProvider(parentRpc);
+
+  let childReceipt: TransactionReceipt | null;
+  try {
+    childReceipt = await childProvider.getTransactionReceipt(trnxHash);
+  } catch (error) {
+    throw new Error(`Failed to get child transaction receipt: ${error.message}`);
+  }
+  if (!childReceipt) {
+    throw new Error(`Transaction receipt not found for hash: ${trnxHash}`);
+  }
+  const messageReceipt = new ChildTransactionReceipt(childReceipt);
+  const parentSigner: Signer = new Wallet(PRIVATE_KEY, parentProvider);
+  const messages = await messageReceipt.getChildToParentMessages(parentSigner);
+  const childToParentMessage = messages[0];
+  if (!childToParentMessage) {
+    console.error("No child-to-parent messages found");
+  }
+  try {
+    const status = await childToParentMessage.status(childProvider);
+    return status;
+  } catch (error) {
+    console.error(`Failed to get message status: ${error.message}`);
+    return undefined;
+  }
+}
```

Comment:
@fcanela The code which calls this function ensures that the message exists. It's not a case where we are uncertain; we always expect to retrieve the message status unless there's an error from the sdk call. It would be more wise to throw an error here?